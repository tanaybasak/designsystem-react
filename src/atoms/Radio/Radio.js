import React from 'react';
import PropTypes from 'prop-types';
import prefix from '../../settings';
import Tooltip from '../Tooltip';

const Radio = ({
  className,
  labelText,
  onChange,
  tooltipTitle,
  tooltipDirection,
  ...restProps
}) => {
  const classnames = `${prefix}-radio-item ${className}`.trim();
  const handleChange = evt => {
    onChange(evt.target.value, evt);
  };
  return (
    <div className={classnames}>
      <input
        className={`${prefix}-radio`}
        type="radio"
        {...restProps}
        onChange={handleChange}
      />
      {labelText ? (
        tooltipTitle ? (
          <Tooltip content={tooltipTitle} direction={tooltipDirection}>
            <label className={`${prefix}-radio-label`} htmlFor={restProps.id}>
              {labelText}
            </label>
          </Tooltip>
        ) : (
          <label className={`${prefix}-radio-label`} htmlFor={restProps.id}>
            {labelText}
          </label>
        )
      ) : null}
    </div>
  );
};

Radio.propTypes = {
  /** Custom class on the Radio wrapper. */
  className: PropTypes.string,
  /** Disable Radio */
  disabled: PropTypes.bool,
  /** Text for Radio Label. */
  labelText: PropTypes.string,
  /** Unique identifier of the element. */
  id: PropTypes.string,
  /** Accepts event handler as prop/argument. */
  /** Tooltip Text for Checkbox Label. */
  tooltipTitle: PropTypes.string,
  /** Tooltip Direction eg: top, bottom, left, right */
  tooltipDirection: PropTypes.oneOf(['top', 'bottom', 'left', 'right']),
  /** Event to subscribe when the Input field is clicked.
   *
   *
   * @value : radio value
   * @event : on change event
   */
  onChange: PropTypes.func,
  /** Control Checked state for Radio */
  checked: PropTypes.bool,
  /** Name of radio element */
  name: PropTypes.string,
  /** Text value to be considered when submitting forms. */
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};

Radio.defaultProps = {
  className: '',
  labelText: '',
  disabled: false,
  tooltipTitle: null,
  tooltipDirection: 'bottom',
  onChange: () => {}
};

export default Radio;
