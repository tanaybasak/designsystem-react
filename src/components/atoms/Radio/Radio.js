import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { prefix } from '../../../settings';

export default function Radio({ className, helperText, checked, ...checkboxProps }) {
    const [isChecked, setValue] = useState(checked || false);

    return (
        <input
            className={`${prefix}-radio ${className}`}
            type="radio"
            checked={isChecked}
            {...checkboxProps}
            onChange={event => {
                setValue(event.currentTarget.checked);
                checkboxProps.onChange(event);
            }}
        />
    )
};

Radio.propTypes = {
    className: PropTypes.string.isRequired,
    disabled: PropTypes.bool,
    id: PropTypes.string,
    onChange: PropTypes.func,
    checked: PropTypes.bool,
    helperText: PropTypes.node,
    name: PropTypes.string,
    title: PropTypes.string,
    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};

Radio.defaultProps = {
    className: '',
    disabled: false,
    onChange: () => { },
    checked: false
};