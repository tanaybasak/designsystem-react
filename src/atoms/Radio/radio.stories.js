import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text, boolean, select } from '@storybook/addon-knobs';
//@update-path-build-start
import Radio from './Radio';
import RadioGroup from './RadioGroup';
//@update-path-build-end

const props = {
  radio: () => ({
    disabled: boolean('Disabled', false)
  })
};
const directionOption = {
  Top: 'top',
  Bottom: 'bottom',
  Left: 'left',
  Right: 'right'
};

storiesOf('Components/Radio', module)
  .add(
    'default',
    () => (
      <Radio
        disabled={boolean('Disabled', false)}
        id="radio1"
        labelText={text('Label', 'Radio Label')}
        name="test"
        value="radio1"
      />
    ),
    {
      info: {
        text: `Description About Radio Component`,
        document: ['Radio']
      }
    }
  )
  .add(
    'with tooltip',
    () => (
      <Radio
        disabled={boolean('Disabled', false)}
        tooltipTitle={text('Tooltip Title', 'Radio Label')}
        tooltipDirection={select(
          'Tooltip Direction',
          directionOption,
          'bottom'
        )}
        id="radio1"
        labelText={text('Label', 'Radio Label')}
        name="test"
        value="radio1"
      />
    ),
    {
      info: {
        text: `Description About Radio Component`,
        document: ['Radio']
      }
    }
  )
  .add(
    'vertical',
    () => (
      <RadioGroup
        orientation="vertical"
        onChange={action('RadioGroup-OnChange')}
        defaultSelected="Radio1"
      >
        <Radio
          {...props.radio()}
          id="radio1"
          name="test"
          value="Radio1"
          labelText={text('Label1', 'Radio Label 1')}
        />
        <Radio
          {...props.radio()}
          id="radio2"
          name="test"
          value="Radio2"
          labelText={text('Label2', 'Radio Label 2')}
        />
        <Radio
          {...props.radio()}
          id="radio3"
          name="test"
          labelText={text('Label3', 'Radio Label 3')}
          value="Radio3"
          disabled
        />
      </RadioGroup>
    ),
    {
      info: {
        text: `Description About Radio Group Component`,
        document: ['Radio']
      }
    }
  )
  .add(
    'horizontal',
    () => (
      <RadioGroup
        onChange={action('Radio Group OnChange')}
        defaultSelected="Radio1"
      >
        <Radio
          {...props.radio()}
          id="radio1"
          name="test"
          value="Radio1"
          labelText={text('Label1', 'Radio Label 1')}
        />
        <Radio
          {...props.radio()}
          id="radio2"
          name="test"
          value="Radio2"
          labelText={text('Label2', 'Radio Label 2')}
        />
        <Radio
          {...props.radio()}
          id="radio3"
          name="test"
          value="Radio3"
          labelText={text('Label3', 'Radio Label 3')}
          disabled
        />
      </RadioGroup>
    ),
    {
      info: {
        text: `Description About Radio Group Component`,
        document: ['Radio']
      }
    }
  );
