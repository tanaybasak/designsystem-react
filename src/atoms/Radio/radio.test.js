import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import prefix from '../../settings';
import Radio from './Radio';

describe('<Radio> component', () => {
    it('snapshots/renders Radio correctly', () => {
        const tree = renderer.create(<Radio />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders radio with label', () => {
        const radio = mount((<Radio labelText="defaultradio" />));
        expect(radio.text()).toBe('defaultradio');
    });

    it('renders radio with default classname', () => {
        const radio = mount((<Radio />));
        expect(radio.find(`.${prefix}-radio`).hasClass('hcl-radio')).toBe(true);
    });

    it('renders radio with default unchecked', () => {
        const radio = mount((<Radio />));
        expect(radio.props().checked).toBeFalsy();
    });

    it('renders radio with default checked', () => {
        const radio = mount((<Radio checked />));
        expect(radio.props().checked).toBeTruthy();
    });

    it('on state change, trigger onChange event ', () => {
        const mockcallback = jest.fn();
        const radio = mount((<Radio onChange={mockcallback} />));
        expect(radio.find('input[type="radio"]').props().checked).toEqual(false);
        radio.find('input[type="radio"]').simulate('change', { target: { checked: true } });
        expect(mockcallback.mock.calls.length).toEqual(1);
    });
});