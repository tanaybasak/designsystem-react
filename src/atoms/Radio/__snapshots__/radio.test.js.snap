// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio> component snapshots/renders Radio correctly 1`] = `
<div
  className="hcl-radio-item"
>
  <input
    checked={false}
    className="hcl-radio"
    disabled={false}
    onChange={[Function]}
    type="radio"
  />
</div>
`;
